Tests/TuistGeneratorTests/Generator/WorkspaceDescriptorGeneratorTests.swift

```diff
@@ -123,6 +123,66 @@ final class WorkspaceDescriptorGeneratorTests: TuistUnitTestCase {
         ])
     }
 
+    func test_generateWorkspaceStructure_withSettingsDescriptorDisablingSchemaGeneration() throws {
+        // Given
+        let temporaryPath = try self.temporaryPath()
+
+        let workspace = Workspace.test(
+            xcWorkspacePath: temporaryPath.appending(component: "Test.xcworkspace"),
+            projects: [],
+            generationOptions: [.automaticSchemeGeneration(.disabled)]
+        )
+
+        let graph = Graph.test(workspace: workspace)
+        let graphTraverser = GraphTraverser(graph: graph)
+
+        // When
+        let result = try subject.generate(graphTraverser: graphTraverser)
+
+        // Then
+        XCTAssertEqual(result.workspaceSettingsDescriptor, WorkspaceSettingsDescriptor(automaticSchemeGeneration: false))
+    }
+
+    func test_generateWorkspaceStructure_withSettingsDescriptorEnablingSchemaGeneration() throws {
+        // Given
+        let temporaryPath = try self.temporaryPath()
+
+        let workspace = Workspace.test(
+            xcWorkspacePath: temporaryPath.appending(component: "Test.xcworkspace"),
+            projects: [],
+            generationOptions: [.automaticSchemeGeneration(.enabled)]
+        )
+
+        let graph = Graph.test(workspace: workspace)
+        let graphTraverser = GraphTraverser(graph: graph)
+
+        // When
+        let result = try subject.generate(graphTraverser: graphTraverser)
+
+        // Then
+        XCTAssertEqual(result.workspaceSettingsDescriptor, WorkspaceSettingsDescriptor(automaticSchemeGeneration: true))
+    }
+
+    func test_generateWorkspaceStructure_withSettingsDescriptorDefaultSchemaGeneration() throws {
+        // Given
+        let temporaryPath = try self.temporaryPath()
+
+        let workspace = Workspace.test(
+            xcWorkspacePath: temporaryPath.appending(component: "Test.xcworkspace"),
+            projects: [],
+            generationOptions: [.automaticSchemeGeneration(.default)]
+        )
+
+        let graph = Graph.test(workspace: workspace)
+        let graphTraverser = GraphTraverser(graph: graph)
+
+        // When
+        let result = try subject.generate(graphTraverser: graphTraverser)
+
+        // Then
+        XCTAssertEqual(result.workspaceSettingsDescriptor, WorkspaceSettingsDescriptor(automaticSchemeGeneration: nil))
```

Comment:
is this correct? I may have misread the code above, would have thought 

```swift
XCTAssertNil(result.workspaceSettingsDescriptor)
```

Reply:
Yes, I believe this is correct. The only time the settings descriptor will be nil is when the `generationOptions` is empty. Otherwise, we treat it as an explicit desire by the user to create the file with the specified value. In this case, it would simply be a near empty XML file. I think the behavior makes more sense in the context of future options being supported where we would not gate the creation of the descriptor on this enum/associated value. Does that make sense?